internal/ingress/controller/parser/parser.go

```diff
@@ -590,6 +623,70 @@ func overrideRoute(route *Route,
 	}
 }
 
+// normalizeProtocols prevents users from mismatching grpc/http
+func normalizeProtocols(route *Route) {
+	protocols := route.Protocols
+	var http, grpc bool
+
+	for _, protocol := range protocols {
+		if strings.Contains(*protocol, "grpc") {
+			grpc = true
+		}
+		if strings.Contains(*protocol, "http") {
+			http = true
+		}
+		if sanitizeProtocol(*protocol) != true {
+			http = true
+		}
+	}
+
+	if grpc && http {
+		route.Protocols = kong.StringSlice("http", "https")
+	}
+	// length check
+	if len(route.Protocols) < 1 {
+		return
+	}
+}
+
+// sanitizeProtocol returns a bool of whether string is a valid protocol
+func sanitizeProtocol(protocol string) bool {
```

Comment:
`sanitize` verb stands for making a change to clean it up I guess. A more appropriate name for the function will be `validProtocol` or `validateProtocol` since all this function is doing is making sure it is a valid protocol.

Reply:
good point. I wasn't wild about the naming.